src/exceptions.py

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Exceptions used by the Synapse charm."""
+
+
+class CharmDatabaseRelationNotFoundError(Exception):
```

Comment:
Perhaps this would be better placed in database_client.py

Reply:
Related: ISD-1070